internal/v2/controller/http/callback.go

```diff
@@ -0,0 +1,87 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"net/http"
+
+	sdkCommon "github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/v2/application"
+	"github.com/gorilla/mux"
+)
+
+func (c *V2HttpController) DeleteDevice(writer http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	id := vars[sdkCommon.IdVar]
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	err := application.DeleteDevice(id, c.dic)
+	if err == nil {
+		c.sendResponse(writer, request, sdkCommon.APIV2DeviceCallbackIdRoute, nil, http.StatusOK)
+	} else {
+		c.sendEdgexError(writer, request, err, sdkCommon.APIV2DeviceCallbackIdRoute, correlationID)
+	}
+}
+
+func (c *V2HttpController) AddUpdateDevice(writer http.ResponseWriter, request *http.Request) {
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	err := application.AddUpdateDevice(request, c.dic)
```

Comment:
here is where you need to first decode the request and determine if it is an Add or Update before calling the App Layer.